shared-ui/src/main/java/org/hzontal/shared_ui/utils/DialogUtils.kt

```diff
@@ -0,0 +1,77 @@
+import android.view.View
+import android.view.ViewGroup
+import android.widget.FrameLayout
+import androidx.core.content.ContextCompat
+import org.hzontal.shared_ui.R
+import org.hzontal.shared_ui.databinding.LayoutBottomMessageBinding
+import org.hzontal.shared_ui.databinding.LayoutBottomMessageWithButtonBinding
+
+object DialogUtils {
+
+    @JvmStatic
+    fun showBottomMessage(context: Activity, msg: String, isError: Boolean, duration: Long = 2000L) {
+        context?.let { showBottomMessage(it, msg, if (isError) R.color.wa_red_error else R.color.tigers_eye, duration) }
+    }
+
+    @JvmStatic
+    fun showBottomMessage(context: Activity, msg: String, isError: Boolean) {
+        context?.let { showBottomMessage(it, msg, if (isError) R.color.wa_red_error else R.color.tigers_eye, 2000L) }
+    }
+
+    @JvmStatic
+    private fun showBottomMessage(context: Activity, msg: String, colorRes: Int, duration: Long) {
+        val binding = LayoutBottomMessageBinding.inflate(LayoutInflater.from(context))
+
+        val container = context.findViewById<ViewGroup>(android.R.id.content)
+        binding.txvMsg.text = msg
+        container.addView(binding.root)
+        binding.root.requestFocus()
+        binding.root.announceForAccessibility(msg)
+        binding.root.alpha = 0f
+        binding.root.animate().alphaBy(1f).setDuration(500).withEndAction {
+            if (binding.root.isAttachedToWindow) {
+                binding.root.animate().alpha(0f).setStartDelay(duration).duration = 500
+            }
+        }
+    }
+
+    @JvmStatic
+    fun showBottomMessageWithButton(context: Activity, msg: String, onBtnClick: () -> Unit) {
+        val container = context.findViewById<ViewGroup>(android.R.id.content)
+
+        // Create a FrameLayout to hold both the overlay and the dialog content
+        val frameLayout = FrameLayout(context)
+        frameLayout.layoutParams = ViewGroup.LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.MATCH_PARENT)
+
+        // Create a transparent background overlay
+        val overlay = View(context)
+        overlay.setBackgroundColor(ContextCompat.getColor(context, android.R.color.black))
+        overlay.alpha = 0.5f // Adjust the alpha value as needed
+        overlay.layoutParams = ViewGroup.LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.MATCH_PARENT)
+        frameLayout.addView(overlay)
+
+        // Create a transparent view to disable interaction with the views behind
+        val disableView = View(context)
+        disableView.setBackgroundColor(ContextCompat.getColor(context, android.R.color.transparent))
+        disableView.isClickable = true
+        disableView.layoutParams = ViewGroup.LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.MATCH_PARENT)
+        frameLayout.addView(disableView)
+        val binding = LayoutBottomMessageWithButtonBinding.inflate(LayoutInflater.from(context), container, false)
```

Comment:
Yellow text against white background : In snackbars the yellow button text should change from the primary yellow (#D6933B) to "yellow/darker" (#8B4F00)